conan/tools/cmake/cmakedeps2/cmakedeps.py

```diff
@@ -70,6 +70,13 @@ def _content(self):
                 direct_deps.append((require, dep))
             config = ConfigTemplate2(self, dep)
             ret[config.filename] = config.content()
+            extra_file_names = (self.get_property("cmake_extra_file_names", dep, check_type=list)
+                                or [])
+            old = self.get_cmake_filename(dep)
+            for f in extra_file_names:
+                filename = f"{f}-config.cmake" if f == f.lower() else f"{f}Config.cmake"
+                ret[filename] = f'include(CMakeFindDependencyMacro)\n' \
+                                f"find_dependency({old})"
```

Comment:
Are we sure the ``find_dependency()`` approach is the best one?
Why not:
- Use a plain ``include(config.filename)``?
- Why not duplicating the contents of ``config.content()`` in each file?

I am a bit concerned the extra ``find_dependency()`` might have corner cases, it might kind of change the actual dependency graph of cmake files?

Reply:
cc @jcar87 who suggested this approach